src/creosote/parsers.py

```diff
@@ -419,3 +419,80 @@ def get_excluded_deps_which_are_not_installed(
 
 def canonicalize_module_name(module_name: str) -> str:
     return module_name.replace("-", "_").replace(".", "_").strip()
+
+
+class DjangoSettingsVisitor(ast.NodeVisitor):
+    """Visit `ast` nodes and extract module names from `INSTALLED_APPS` and `MIDDLEWARE`."""
+
+    def __init__(self) -> None:
+        self.found_modules: List[str] = []
+        self.variable_assignments: dict[str, list[str]] = {}
+
+    def visit(self, node: ast.AST) -> None:
+        """Walk the tree to find all list assignments first."""
+        for sub_node in ast.walk(node):
+            if isinstance(sub_node, ast.Assign):
+                self.visit_Assign_for_list(sub_node)
+        super().visit(node)
+
+    def visit_Assign_for_list(self, node: ast.Assign) -> None:
+        """Find all list/tuple assignments and store their values."""
+        if isinstance(node.value, (ast.List, ast.Tuple)):
+            apps = []
+            for element in node.value.elts:
+                if isinstance(element, ast.Constant) and isinstance(element.value, str):
+                    apps.append(element.value.split(".")[0])
+            for target in node.targets:
+                if isinstance(target, ast.Name):
+                    self.variable_assignments[target.id] = apps
+
+    def _resolve_value(self, node: ast.expr) -> list[str]:
+        """Recursively resolve the value of a node."""
+        if isinstance(node, (ast.List, ast.Tuple)):
+            apps = []
+            for element in node.elts:
+                if isinstance(element, ast.Constant) and isinstance(element.value, str):
+                    apps.append(element.value.split(".")[0])
+            return apps
+        elif isinstance(node, ast.Name) and node.id in self.variable_assignments:
+            return self.variable_assignments[node.id]
+        elif isinstance(node, ast.BinOp) and isinstance(node.op, ast.Add):
+            left = self._resolve_value(node.left)
+            right = self._resolve_value(node.right)
+            return left + right
+        return []
```

Comment:
The recursive `_resolve_value` method could infinite loop if `settings.py` has circular variable references. Suggestion:


```suggestion
		 def _resolve_value(self, node: ast.expr, visited: set[str] | None = None) -> list[str]:
		     """Recursively resolve the value of a node."""
		     if visited is None:
		         visited = set()
		
		     if isinstance(node, (ast.List, ast.Tuple)):
            apps = []
            for element in node.elts:
                if isinstance(element, ast.Constant) and isinstance(element.value, str):
                    apps.append(element.value.split(".")[0])
            return apps
		     elif isinstance(node, ast.Name):
		         if node.id in visited:
		             logger.warning(f"Circular variable reference detected: {node.id}")
		             return []
		         if node.id in self.variable_assignments:
		             visited.add(node.id)
		             return self.variable_assignments[node.id]
		     elif isinstance(node, ast.BinOp) and isinstance(node.op, ast.Add):
		         left = self._resolve_value(node.left, visited)
		         right = self._resolve_value(node.right, visited)
		         return left + right
		     return []
```